### PR TITLE
Adding testing to pixelflow

### PR DESCRIPTION
**Pull request by IFenton**
Addressing #49

### CODE REVIEW

**Comment by IFenton** [comment]
I've added a series of tests for basic functionality in pixelflow, resolving #49

File: src/pixelflow/core.py [hunk 1]
```diff
@@ -208,12 +208,26 @@ def pixelflow(
...
+            features = (
+                "label",
+                "bbox",
+                "centroid",
+                "area",
+                "major_axis_length",
+                "eccentricity",
+                "orientation",
+                "solidity",
+            )
```
Comment (quantumjot) [inline src/pixelflow/core.py:230]: Is there a reason to specify these here?

File: src/pixelflow/core.py [hunk 2]
```diff
@@ -208,12 +208,26 @@ def pixelflow(
         else:
             spacing = (1,) * mask.ndim
 
+    if any(val == 0 for val in spacing):
+        raise ValueError("Spacing cannot be zero")
```
Comment (quantumjot) [inline src/pixelflow/core.py:212]: Might be good to return the spacing in the error message to help with debugging?

```suggestion
        raise ValueError(f"Spacing cannot be zero: {spacing}")
```

File: tests/conftest.py [hunk 3]
```diff
@@ -9,11 +9,16 @@
...
+    num_blobs = 6
     img, coords = simulate_image(size=size, num_blobs=num_blobs)
-    mask = label(img > 2.3)
+    mask = label(img > 2.2)
+    # remove small objects
+    tmp = np.unique(mask, return_counts=True)
+    if any(tmp[1] < 5):
+        for val in tmp[0][tmp[1] < 5]:
+            mask[mask == val] = 0
+        mask = label(mask)
```
Comment (quantumjot) [inline tests/conftest.py:21]: You could consider using this:
https://scikit-image.org/docs/stable/api/skimage.morphology.html#skimage.morphology.remove_small_objects

**Review by quantumjot — APPROVED** [review]
Looks good - nice to see all the new tests!

A few comments/suggestions below

File: src/pixelflow/core.py (hunk 1 above)
Comment (IFenton) [reply src/pixelflow/core.py:230]: At the moment, if you don't supply any features for a 2D image, then you get an error, however a 3D image gives you subset of the possible 3D features (which is the regionprops3D default). So to give consistency I added a set of the common features for 2D, all of which work with isotropic and anisotropic spacing. If there's any particular ones that you think should or shouldn't be included then happy to alter the list.

File: tests/conftest.py (hunk 3 above)
Comment (IFenton) [reply tests/conftest.py]: Done

File: src/pixelflow/core.py (hunk 1 above)
Comment (quantumjot) [reply src/pixelflow/core.py:230]: Cool - thanks for the explanation!

**Pull request merged**